src/Uno.UI/Microsoft/UI/Xaml/Controls/TreeView/TreeViewNodeVector.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Uno.Extensions;
+using Uno.UI;
+using Windows.Foundation.Collections;
+
+namespace Microsoft.UI.Xaml.Controls
+{
+	internal class TreeViewNodeVector : ObservableVector<TreeViewNode>
+	{
+		private TreeViewNode m_parent;
+
+		public TreeViewNodeVector()
+		{
+
+		}
+
+		public void SetParent(TreeViewNode value)
+		{
+			m_parent = value;
+		}
+
+		private IList GetWritableParentItemsSource()
+		{
+			IList parentItemsSource = null;
+
+			if (m_parent?.ItemsSource != null)
+			{
+				parentItemsSource = m_parent?.ItemsSource as IList;
+			}
+
+			return parentItemsSource;
+		}
+
+		public void Append(TreeViewNode item, bool updateItemsSource = true)
+		{
+			InsertAt(Count, item, updateItemsSource);
+		}
+
+		public override void Add(TreeViewNode item) => Append(item);
+
+		public void InsertAt(int index, TreeViewNode item, bool updateItemsSource = true)
+		{			
+			if (m_parent == null)
+			{
+				throw new InvalidOperationException("Parent node must be set");
+			}
+			if (index > base.Count)
+			{
+				throw new IndexOutOfRangeException("Index out of range for Insert");
```

Comment:
Codacy found an issue: ['System.IndexOutOfRangeException' should not be thrown by user code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)